### PR TITLE
DOC: Upgraded Docstring pandas.DataFrame.dot

### PR DESCRIPTION
**Pull request by HubertKl**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -953,16 +953,61 @@ def __len__(self):
...
+            The other object to compute the matrix product with.
 
         Returns
         -------
-        dot_product : DataFrame or Series
+        Series or DataFrame
+            Return the Series of the matrix product between the Dataframe and
+            other if other is a Series, the Dataframe of the matrix product of
+            each columns of the DataFrame/np.array and each columns of other
+            if other is a DataFrame or a numpy.ndarray.
```
Comment (datapythonista) [inline pandas/core/frame.py]: Can you rephrase this? Can't really understand it, we probably need to have more than one sentence.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -953,16 +953,61 @@ def __len__(self):
...
+
+        See Also
+        --------
+        Series.dot: Compute the inner product of a Series with another Series
+            or the columns of a DataFrame or the columns of a numpy array.
```
Comment (datapythonista) [inline pandas/core/frame.py]: Can you say instead "Same method for Series." or similar if that makes sense.

File: pandas/core/frame.py [hunk 3]
```diff
@@ -953,16 +953,61 @@ def __len__(self):
...
+        The dimensions of DataFrame and other must be compatible in order to
+        compute the matrix multiplication.
+
+        The dot method for Series computes the inner product, instead of the
+        matrix product here.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame([[0, 1, -2, 3], [4, -5, 6, 7]])
+        >>> s = pd.Series([0, 1, 2, 3])
```
Comment (datapythonista) [inline pandas/core/frame.py]: Can you try to use smaller  / easier values, so the math can be done mentally?

Also, avoid the `print` functions, the value will be shown without them too.

And divide the examples in groups with a blank line between them, and probably a short explanation of what you're doing would help reading.

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution @HubertKl, and sorry this got lost and wasn't reviewed until now.

Added few comments.

**Comment by datapythonista** [comment]
@HubertKl do you have time to address the review comments and fix the conflicts?

**Comment by HubertKl** [comment]
I tried to address the issues. Sorry for the delay but I didn't figure that the review would append only now.

File: pandas/core/frame.py [hunk 4]
```diff
@@ -1046,16 +1046,71 @@ def __len__(self):
...
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects. Can also be
-        called using `self @ other` in Python >= 3.5.
+        Compute the matrix mutiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series, DataFrame or a numpy array.
+
+        It can also be called using `self @ other` in Python >= 3.5.
```
Comment (datapythonista) [inline pandas/core/frame.py]:
```suggestion
        It can also be called using ``self @ other`` in Python >= 3.5.
```

sorry I didn't see that before, for code we use double backticks (single backticks are for things we reference, like a function name, a class, a variable...)

File: pandas/core/frame.py (hunk 2 above)
Comment (datapythonista) [reply pandas/core/frame.py]: do you mind changing this?

**Review by datapythonista — COMMENTED** [review]
looks great, just two minor things, and it's ready

**Comment by HubertKl** [comment]
It is done.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @HubertKl 

if you can merge master and update the PR branch, to rerun the CI and leave it green

**Comment by datapythonista** [comment]
@HubertKl do you have time to merge master into your branch, so we can merge?

**Comment by datapythonista** [comment]
@jreback this should be ready to be merged

**Comment by jreback** [comment]
thanks @HubertKl
